generator/src/main/java/org/jdbi/v3/generator/GenerateProcessor.java

```diff
@@ -0,0 +1,139 @@
+    }
+
+    private void generate(Element e) {
+        try {
+            generate0(e);
+        } catch (Exception ex) {
+            processingEnv.getMessager().printMessage(Kind.ERROR, "Failure: " + ex, e);
+            throw new RuntimeException(ex);
+        }
+    }
+
+    final Set<ElementKind> ACCEPTABLE = EnumSet.of(ElementKind.CLASS, ElementKind.INTERFACE);
+    private void generate0(Element e) throws IOException {
+        processingEnv.getMessager().printMessage(Kind.WARNING, String.format("[jdbi] generating for %s", e));
+        if (!ACCEPTABLE.contains(e.getKind())) {
+            throw new IllegalStateException("Generate on non-class: " + e);
+        }
+
+        final TypeElement te = (TypeElement) e;
+        final String implName = te.getSimpleName() + "Impl";
+        final JavaFileObject file = processingEnv.getFiler().createSourceFile(implName, e);
+        try (Writer out = file.openWriter()) {
+            final TypeSpec.Builder builder = TypeSpec.classBuilder(implName);
+            final TypeName superName = TypeName.get(te.asType());
+            if (te.getKind() == ElementKind.CLASS) {
+                builder.superclass(superName);
+            } else {
+                builder.addSuperinterface(superName);
+            }
+            builder.addSuperinterface(SqlObject.class);
+            builder.addField(InvocationHandler.class, "handler", Modifier.FINAL, Modifier.PRIVATE);
+            builder.addMethod(MethodSpec.constructorBuilder()
+                    .addParameter(InvocationHandler.class, "handler")
+                    .addCode("this.handler = handler;\n")
+                    .build());
+
+            builder.addMethod(generateMethod(builder, getHandle()));
+
+            te.getEnclosedElements().stream()
+                .filter(ee -> ee.getKind() == ElementKind.METHOD)
+                .filter(ee -> ee.getModifiers().contains(Modifier.ABSTRACT))
+                .map(ee -> generateMethod(builder, ee))
+                .forEach(builder::addMethod);
+
+            JavaFile.builder(packageName(te), builder.build()).build().writeTo(out);
+        }
+    }
+
+    private MethodSpec generateMethod(TypeSpec.Builder typeBuilder, Element e) {
+        final ExecutableElement ee = (ExecutableElement) e;
+        final Builder builder = MethodSpec.overriding(ee);
+        final String params = ee.getParameters().stream().
+                map(VariableElement::getSimpleName)
+                .collect(Collectors.joining(","));
+        String methodField = "_" + e.getSimpleName() + "Method";
+        typeBuilder.addField(Method.class, methodField, Modifier.PRIVATE, Modifier.FINAL, Modifier.STATIC);
+        typeBuilder.addStaticBlock(CodeBlock.of("$L = $L.class.getMethod($S, $L);\n",
+                methodField,
+                ee.getEnclosingElement().getSimpleName(),
+                e.getSimpleName(),
+                ee.getParameters().stream().map()));
+        processingEnv.getElementUtils().
```

Comment:
what's happening here?

Reply:
This creates a static field for each method and a static block that fills it with its corresponding `java.lang.reflect.Method` instance, so we can easily dispatch to the existing invocation handler without making changes..

Reply:
Now we use a `HandlerKey` class instead of the `Method` objects.  That way we don't have any reflective types at all.